components/radio-button/script.js

```diff
@@ -248,31 +247,26 @@ class RadioButton extends BaseComponent {
         }
     }
 
-    /**
-     * Initialize the custom component.
-     * Set template, attach event listeners, setup initial state etc.
-     * @param {object} data
-    */
-    init(data) {
-        const radioButtonText = this.textContent;
-
-        this.setupTemplate(data, () => {
-            components.renderOnce(this);
-
-            this.textElement = this.querySelector('.radio-button-text');
-            // Apply the user set text.
-            this.textElement.textContent = radioButtonText;
-            if (this.disabled) this.firstChild.classList.add('guic-radio-button-disabled');
-        });
-    }
-
     // eslint-disable-next-line require-jsdoc
     connectedCallback() {
         // Get the text set from the user before applying the template.
         this.radioGroup = this.parentElement;
 
+        let radioButtonText;
+        const hasSlots = this.querySelectorAll('[slot="radio-button-content"]').length;
+        // use text content if there is no slot
+        if (!hasSlots) radioButtonText = this.textContent;
+
         components.loadResource(this)
-            .then(this.init)
+            .then((data) => {
+                this.setupTemplate(data, () => {
+                    components.renderOnce(this);
+                    this.textElement = this.querySelector('.radio-button-text');
+                    if (!hasSlots) this.textElement.textContent = radioButtonText;
+                    // Apply the user set text
+                    if (this.disabled) this.firstChild.classList.add('guic-radio-button-disabled');
+                });
+            })

```

Comment:
Why did we move the code here and removed the init method? :D

Reply:
I wanted to access the html before render, and create variable that I can use in init and I didn't want to make it a member, so I moved init as an arrow function, but now I see that I don't have to do it outside init, I can create the vat before components.renderOnce.